### PR TITLE
Register multiple classes at once

### PR DESCRIPTION
**Pull request by tanndlin**

### CODE REVIEW

**Review by c650 — APPROVED** [review]

File: src/app/plugins/register.plugin.ts [hunk 1]
```diff
@@ -17,16 +17,34 @@ export class RegisterPlugin extends Plugin {
...
-    } catch (e) {
-      message.reply(e);
+
+    const successfulClasses: string[] = [];
+    const invalidClasses: string[] = [];
+    for (const arg of args) {
+      const request = this.container.classService.buildRequest(message.author, [arg]);
+      if (!request) {
+        message.reply('I was unable to build your request.');
+        return;
```
Comment (joey-colon) [inline src/app/plugins/register.plugin.ts:30]: `continue;`

**Review by joey-colon — APPROVED** [review]

File: src/app/plugins/register.plugin.ts [hunk 2]
```diff
@@ -17,16 +17,34 @@ export class RegisterPlugin extends Plugin {
...
-      message.reply(response);
-    } catch (e) {
-      message.reply(e);
+
+    const successfulClasses: string[] = [];
+    const invalidClasses: string[] = [];
+    for (const arg of args) {
+      const request = this.container.classService.buildRequest(message.author, [arg]);
+      if (!request) {
+        message.reply('I was unable to build your request.');
```
Comment (joey-colon) [inline src/app/plugins/register.plugin.ts:29]: also might not need this.

File: src/app/plugins/register.plugin.ts [hunk 3]
```diff
@@ -17,16 +17,34 @@ export class RegisterPlugin extends Plugin {
...
+          successfulClasses.push(arg);
+        } else {
+          invalidClasses.push(arg);
+        }
+      } catch (e) {
+        message.reply(e);
+      }
+    }
+
+    let messageForUser = `Successfully added to ${successfulClasses.length} classes`;
```
Comment (joey-colon) [inline src/app/plugins/register.plugin.ts:44]: then u will need to handle the cases of when no classes were added successfully (just have specific message if len == 0)

File: src/app/plugins/register.plugin.ts (hunk 2 above)
Comment (c650) [reply src/app/plugins/register.plugin.ts:29]: wouldnt want this to be sent n times

Comment (tanndlin) [reply src/app/plugins/register.plugin.ts:29]: So check if it failed for each arg, then reply to the user, outside of the for loop, that one or more requests failed?

Comment (c650) [reply src/app/plugins/register.plugin.ts:29]: include it in the array of classes that could not be added (no 'success')

**Review by joey-colon — APPROVED** [review]

File: src/app/plugins/register.plugin.ts [hunk 4]
```diff
@@ -17,16 +17,40 @@ export class RegisterPlugin extends Plugin {
...
+      try {
+        const response = await this.container.classService.register(request);
+        if (response.includes('success')) {
+          successfulClasses.push(arg);
+        } else {
+          invalidClasses.push(arg);
+        }
+      } catch (e) {
+        message.reply(e);
+      }
```
Comment (c650) [inline src/app/plugins/register.plugin.ts:41]: we dont want this to happen n times.

Comment (tanndlin) [reply src/app/plugins/register.plugin.ts:41]: Should I just push the arg to `invalidClasses` and `this.container.loggerService.error(e)`?

File: src/app/plugins/register.plugin.ts [hunk 5]
```diff
@@ -17,16 +17,46 @@ export class RegisterPlugin extends Plugin {
...
+      try {
+        const response = await this.container.classService.register(request);
+        if (response.includes('success')) {
+          successfulClasses.push(arg);
+        } else {
+          invalidClasses.push(arg);
+        }
+      } catch (e) {
+        invalidClasses.push(arg);
+        error = e;
```
Comment (c650) [inline src/app/plugins/register.plugin.ts:42]: just log the error with logger service. dont expose the actual error to the user

Comment (tanndlin) [reply src/app/plugins/register.plugin.ts:42]: I can, but before, it just replied to the user with the error

Comment (c650) [reply src/app/plugins/register.plugin.ts:42]: you shouldnt expose the actual exception to the user

Comment (tanndlin) [reply src/app/plugins/register.plugin.ts:42]: Ok, should I log each error just in case, or just do one log with an error per call

Comment (c650) [reply src/app/plugins/register.plugin.ts:42]: just log the error when it occurs that's all

**Review by c650 — APPROVED** [review]
:confetti_ball:

**Pull request merged**